server/src/main/java/org/cloudfoundry/identity/uaa/provider/VmidentityIdentityProviderProvisioning.java

```diff
@@ -80,8 +81,8 @@ public IdentityProvider retrieve(String id) {
     public List<IdentityProvider> retrieveAll(boolean activeOnly, String zoneId) {
         try {
             // TODO: active only - we only have active right now...
-            String tenant = VmidentityUtils.getTenantName(zoneId, this._idmClient.getSystemTenant());
-            Collection<IIdentityStoreData> providers = this._idmClient.getProviders(
+            String tenant = VmidentityUtils.getTenantName(zoneId, this.client.getSystemTenant());
```

Comment:
i'm not sure this approach will work. we seem to map proider id, to idm's provider name. Yet we put domain name for user origin. i think this won;t wotk. provider name != [provider domain, and more over some providers have multiple domains. why do we not unify on provider is = idm's provider anme ? this seems easier to handle ....